Foundation/Collection/Collection.hs

```diff
@@ -71,17 +71,18 @@ class (IsList c, Item c ~ Element c) => Collection c where
     -- | Check if a collection contains a specific element
     --
     -- This is the inverse of `notElem`.
-    elem :: Eq (Element c) => Element c -> c -> Bool
+    elem :: forall a . (Eq a, a ~ Element c) => Element c -> c -> Bool
```

Comment:
@vincenthz 
I'm curious about those changes; can you explain the reason behind this ?